File: PyLibs/src/Libs/MsgRoute/MsgRouteNotify.py
# -*- coding: utf-8 -*-

from Libs.Core.GObject import GObject;
from Libs.DataStruct.MList import MList;

class MsgRouteNotify(GObject):

    def __init__(self):
        super(MsgRouteNotify, self).__init__();

        self.mTypeId = "MsgRouteNotify";
        
        self.mDispList = MList();
 

    def addOneDisp(self, disp):   
        if(self.mDispList.IndexOf(disp) == -1):
            self.mDispList.Add(disp);



    def emoveOneDisp(self, disp):
        if(self.mDispList.IndexOf(disp) != -1):
            self.mDispList.Remove(disp);


    def handleMsg(self, msg):
        for item in self.mDispList:
            item.handleMsg(msg);

File: JSLib/src/MsgRoute/MsgRouteNotify.js
using System.Collections.Generic;

namespace SDK.Lib
{
    public class MsgRouteNotify
    {
        protected List<MsgRouteDispHandle> m_dispList = new List<MsgRouteDispHandle>();

        public void addOneDisp(MsgRouteDispHandle disp)
        {
            if(m_dispList.IndexOf(disp) == -1)
            {
                m_dispList.Add(disp);
            }
        }

        public void removeOneDisp(MsgRouteDispHandle disp)
        {
            if(m_dispList.IndexOf(disp) != -1)
            {
                m_dispList.Remove(disp);
            }
        }

        public void handleMsg(MsgRouteBase msg)
        {
            foreach(var item in m_dispList)
            {
                item.handleMsg(msg);
            }

            Ctx.m_instance.m_poolSys.deleteObj(msg);
        }
    }
}